### PR TITLE
feat: preprocessing and data generation

### PR DESCRIPTION
**Pull request by jupyterjazz**
Closes https://github.com/jina-ai/now/issues/417 and https://github.com/jina-ai/now/issues/421

Implements preprocessing and fine-tuning data generation (part of setup) for text to text+image app.

- For the moment, we don't ask user for anything and define all required parameters in `resources/nest/config.json` which is loaded as a `Task` dataclass.
- Data generation is general (not only for nest). It gets a dataset and information about encoder + generation methods and produces fine-tuning data.

### CODE REVIEW

File: tests/resources/nest/config.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "wordlift-multi-modal-search",
```
Comment (jupyterjazz) [inline tests/resources/nest/config.json:2]: change dataset name

**Review by florian-hoenicke — APPROVED** [review]
LGTM

**Review by joschkabraun — COMMENTED** [review]
See minor comments. Otherwise LGTM!

File: now/now_dataclasses.py [hunk 2]
```diff
@@ -1,9 +1,87 @@
 from dataclasses import dataclass
-from typing import Dict, List, Optional
+from typing import Any, Dict, List, Optional
+
+from pydantic.dataclasses import dataclass as pydantic_dataclass
```
[the remark this answers is not in the document]
Comment (jupyterjazz) [reply now/now_dataclasses.py:4]: agree. let's stick with pydantic, it's more convenient

**Pull request merged**